content/blog/2020/07/2020-07-30-github-checks-api-plugin-coding-phase-2.adoc

```diff
@@ -0,0 +1,149 @@
+
+1. Update Warnings NG plugin to 8.4.0
+2. Install link:https://plugins.jenkins.io/checks-api/[Checks API plugin] and link:https://plugins.jenkins.io/github-checks/[GitHub Checks plugin] on your Jenkins instance
+3. Use warnings-ng plugin in your Jenkinsfile for the multi-branch project or GitHub organization project, e.g.
+
+[source, groovy]
+----
+node {
+    stage ('Checkout') {
+        checkout scm
+    }
+
+    stage ('Build and Static Analysis') {
+        withMaven {
+            sh 'mvn -V -e clean verify -Dmaven.test.failure.ignore'
+        }
+
+        recordIssues tools: [java(), javaDoc()], aggregatingResults: 'true', id: 'java', name: 'Java'
+        recordIssues tool: errorProne(), healthy: 1, unhealthy: 20
+        recordIssues tools: [checkStyle(pattern: 'target/checkstyle-result.xml'),
+            spotBugs(pattern: 'target/spotbugsXml.xml'),
+            pmdParser(pattern: 'target/pmd.xml'),
+            cpd(pattern: 'target/cpd.xml')],
+            qualityGates: [[threshold: 1, type: 'TOTAL', unstable: true]]
+    }
+}
+----
+
+_For more about the pipeline usage of warnings-ng plugin, please see link:https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#pipeline-configuration[the official documentation]._
+
+However, if you don't want to publish the warnings to GitHub, you can definitely disable it by using a boolean field `skipPublishingChecks`, e.g.
+
+[source, groovy]
+----
+recordIssues enabledForFailure: true, tools: [java(), javaDoc()], skipPublishingChecks: true
+----
+
+== Coverage Checks
+
+The coverage checks are achieved by consuming the API in link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage API plugin].
+First, in the conversation tab of a PR, users will be able to see the summary about the coverage and the comparing with previous builds.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-summary.png[Coverage Summary]
+
+Besides, the `Details` page will contain some other things:
+
+* The links to the reference build, including the target branch build from the master branch and the last successful build from this branch
+* Coverage healthy score (the default value is 100% if the threshold is not configured)
+* Coverages and trends of different types in table format
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-details.png[Coverage Details]
+
+The link:https://github.com/jenkinsci/code-coverage-api-plugin/pull/169[pull request] for this feature will soon be merged and will be included in the next release of Coverage Checks API plugin. After that, you can use it by adding the below section to your pipeline script:
+
+[source, groovy]
+----
+node {
+    stage ('Checkout') {
+        checkout scm
+    }
+
+    stage ('Line and Branch Coverage') {
+        withMaven {
+            sh "mvn -V -U -e jacoco:prepare-agent test jacoco:report -Dmaven.test.failure.ignore"
+        }
```

Comment:
the jacoco step isn't actually needed if you follow the docs in the plugin for configuring maven, https://plugins.jenkins.io/code-coverage-api/ (search jacoco), not sure if you want to rework this a bit for that?
```suggestion
        sh "mvn -V -U -e jacoco:prepare-agent test jacoco:report -Dmaven.test.failure.ignore"
```